runner/src/main.rs

```diff
@@ -23,16 +23,152 @@
 //! cargo run --package=runner
 //! ```
 
-use std::{io::Read, path::PathBuf};
+use std::{
+    io::Read,
+    path::{Path, PathBuf},
+};
 use structopt::StructOpt;
 
 mod internal;
 use internal::*;
 
 fn main() -> Result<(), Box<dyn std::error::Error>> {
     let opt = Opt::from_args();
+
+    let steps = match opt.cmd {
+        Command::BuildExamples(ref opt) => build_examples(&opt),
+        Command::BuildServer(ref opt) => build_server(&opt),
+        Command::RunTests => run_tests(),
+    };
     // TODO(#396): Add support for running individual commands via command line flags.
-    let root = Step::Multiple {
+    let statuses = run_step(&Context::root(&opt), &steps);
+
+    // If the overall status value is an error, terminate with a nonzero exit code.
+    if statuses.contains(&StatusResultValue::Error) {
+        std::process::exit(1);
+    }
+
+    Ok(())
+}
+
+fn build_examples(opt: &BuildExamples) -> Step {
+    match opt.variant.as_str() {
+        "rust" => Step::Multiple {
+            name: "root".to_string(),
+            steps: examples()
+                .filter(is_cargo_toml_file)
+                .filter(|entry| {
+                    (*entry != PathBuf::from("./examples/aggregator/backend/Cargo.toml"))
```

Comment:
Could you please add a comment, why these targets are excluded?

Reply:
Done.